packages/emnapi/src/value/convert2c.ts

```diff
@@ -76,17 +74,13 @@ mergeInto(LibraryManager.library, {
     },
 
     getOrUpdateMemoryView: function<T extends ArrayBufferView> (view: T): T {
-      let bufferInfo: PointerInfo | undefined
-      if (typeof Buffer !== 'undefined' && view instanceof Buffer && emnapiExternalMemory.bufferTable.has(view as any)) {
-        bufferInfo = emnapiExternalMemory.bufferTable.get(view as any)
-      }
       if (view.buffer === HEAPU8.buffer) {
         if (!emnapiExternalMemory.wasmMemoryViewTable.has(view)) {
           emnapiExternalMemory.wasmMemoryViewTable.set(view, {
             Ctor: view.constructor as any,
-            ptr: view.byteOffset,
+            address: view.byteOffset,
             length: ((view) => {
```

Comment:
Now that I'm thinking about it, isn't this function just replicating `view.length` for all listed types except `DataView`? If so, it can be greatly simplified to `length: view instanceof DataView ? view.byteLength : view.length`.

Reply:
nice catch